cmd/common.go

```diff
@@ -35,14 +35,18 @@ func initBinaries() {
 	}
 }
 
+type PubSpec struct {
+	Name         string
+	Description  string
+	Version      string
+	Author       string
+	Dependencies map[string]interface{}
+}
+
 // assertInFlutterProject asserts this command is executed in a flutter project
 // and returns the project name.
-func assertInFlutterProject() string {
+func assertInFlutterProject() PubSpec {
```

Comment:
I feel that the `os.Exit(1)` has become a side-effect of "read and parse pubspec". Especially now that this function is called more often, we may want to refactor. Opening,reading,parsing on every call isn't cheap and not necessary. Either that or we shouldn't call this function so often in the packaging code below (`assertInFlutterProject().Name`, `assertInFlutterProject().Description`, etc.)